src/python_testing/matter_testing_support.py

```diff
@@ -560,6 +692,90 @@ def record_warning(self, test_name: str, location: Union[AttributePathLocation,
     def record_note(self, test_name: str, location: Union[AttributePathLocation, EventPathLocation, CommandPathLocation], problem: str, spec_location: str = ""):
         self.problems.append(ProblemNotice(test_name, location, ProblemSeverity.NOTE, problem, spec_location))
 
+    def on_fail(self, record):
+        ''' Called by Mobly on test failure
+
+            record is of type TestResultRecord
+        '''
+        if self.runner_hook and not self.is_commissioning:
+            exception = record.termination_signal.exception
+            step_duration = (datetime.now(timezone.utc) - self.step_start_time) / timedelta(microseconds=1)
+            # This isn't QUITE the test duration because the commissioning is handled separately, but it's clsoe enough for now
+            # This is already given in milliseconds
+            test_duration = record.end_time - record.begin_time
+            # TODO: I have no idea what logger, logs, request or received are. Hope None works because I have nothing to give
+            self.runner_hook.step_failure(logger=None, logs=None, duration=step_duration, request=None, received=None)
```

Comment:
```suggestion
            self.runner_hook.step_failure(logger=None, logs=record.details, duration=step_duration, request=None, received=None)
```
This should provide some feedback

Reply:
have you tested this already?

Reply:
I tested with TC_ACE_1_3 test case, forcing an assert fail using the code bellow. The `record.details` is filled with the assert message `Returned attribute......`. I think we should also test using other scenarios.
```
asserts.assert_true(False,
                            'Returned attribute {} is wrong type expected {}, got {}'.format(attribute, desired_type, type(attr_ret)))
```

Reply:
ok, I think let's address this as a follow up. The problem here is that the hooks API is not well documented, so I'm not sure what is actually required here. I thought the intent here was to supply logs in possibly the yaml format? The API seems to have stuff from the yaml impl leaking in.

Reply:
Tracking this in #30844, going to close this comment for now so we can get this PR in.